web/models/confirmable.ex

```diff
@@ -90,7 +90,7 @@ defmodule Coherence.Schema.Confirmable do
   deprecated! Please use Coherence.ControllerHelpers.unlock!/1.
   """
   def confirm(user) do
-    Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil})
+    Config.user_schema.changeset(user, %{confirmed_at: Ecto.DateTime.utc, confirmation_token: nil, current_password: user.password})
```

Comment:
The current_password should not be required here. Please update the changeset logic so it only requires current_password validation if the password is being changed.